samples/quickperf/src/test/java/com/google/cloud/jdbc/quickperf/AppTest.java

```diff
@@ -0,0 +1,248 @@
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.InstanceConfigId;
+import com.google.cloud.spanner.InstanceInfo;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.cloud.spanner.connection.SpannerPool;
+import com.google.common.collect.ImmutableList;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import com.google.spanner.admin.database.v1.CreateDatabaseRequest;
+
+import io.grpc.InternalConfigSelector.Result;
+
+import com.google.spanner.admin.instance.v1.CreateInstanceRequest;
+import com.google.spanner.admin.instance.v1.Instance;
+import com.google.spanner.admin.instance.v1.InstanceConfigName;
+import com.google.spanner.admin.instance.v1.ProjectName;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.springframework.boot.SpringApplication;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.exc.StreamReadException;
+import com.fasterxml.jackson.databind.DatabindException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.google.api.gax.longrunning.OperationFuture;
+
+public class AppTest {
+    private static String TEST_FILE = "src/test/resources/testfile.json";
+
+    private static GenericContainer<?> emulator;
+
+    private static String projectId = "test";
+    private static String instanceId = "test";
+    private static String databaseId = "quickperftest";
+
+    private static DatabaseAdminClient dbAdminClient;
+
+    @BeforeClass
+    public static void setup() throws InterruptedException, ExecutionException {
+        System.out.println("Starting Emulator");
+        emulator = new GenericContainer<>(
+                DockerImageName.parse("gcr.io/cloud-spanner-emulator/emulator:latest"))
+                .withExposedPorts(9010)
+                .waitingFor(Wait.forListeningPort());
+
+        emulator.setPortBindings(ImmutableList.of("9010:9010"));
```

Comment:
See my comment further above. You don't have to force the emulator to use port 9010, you can also let it run on a random port, and just add `localhost:<port>` to the connection URL. The `autoConfigEmulator=true` flag will still work as it does now, but will use the host/port in the connection URL instead of the default `localhost:9010`.